test/framework/src/main/java/org/elasticsearch/readiness/MockReadinessService.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.readiness;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.node.MockNode;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.plugins.Plugin;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.SocketAddress;
+import java.net.SocketOption;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.spi.SelectorProvider;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class MockReadinessService extends ReadinessService {
+    /**
+     * Marker plugin used by {@link MockNode} to enable {@link MockReadinessService}.
+     */
+    public static class TestPlugin extends Plugin {}
+
+    private static class MockServerSocketChannel extends ServerSocketChannel {
+        private static final Map<String, MockServerSocketChannel> mockedSockets = new HashMap<>();
```

Comment:
Shouldn't this be a ConcurrentHashMap since many threads could be involved? At minimum the readiness service running in a different thread than the test itself that asserts on the socket state.

Reply:
Good catch, I had a simple map initially and made a mental note to review that.. and forgot. Will fix.